pkg/asset/installconfig/baremetal/baremetal.go

```diff
@@ -0,0 +1,165 @@
+// Package baremetal collects bare metal specific configuration.
+package baremetal
+
+import (
+	"fmt"
+	"gopkg.in/AlecAivazis/survey.v1"
+
+	"github.com/openshift/installer/pkg/types/baremetal"
+	baremetaldefaults "github.com/openshift/installer/pkg/types/baremetal/defaults"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+// Platform collects bare metal specific configuration.
+func Platform() (*baremetal.Platform, error) {
+	var libvirtURI, ironicURI, externalBridge, provisioningBridge, apiVIP, ingressVIP string
+	var hosts []*baremetal.Host
+
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Libvirt Connection URI",
+				Help:    "The libvirt connection URI to be used.",
+				Default: baremetaldefaults.LibvirtURI,
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
+		},
+	}, &libvirtURI)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Ironic Connection URI",
+				Help:    "The ironic connection URI to be used.",
+				Default: baremetaldefaults.IronicURI,
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
+		},
+	}, &libvirtURI)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "External bridge",
+				Help:    "External bridge is used for external communication.",
+				Default: baremetaldefaults.ExternalBridge,
+			},
+			Validate: survey.ComposeValidators(survey.Required, interfaceValidator),
+		},
+	}, &externalBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning bridge",
+				Help:    "Provisioning bridge is used to provision machines.",
+				Default: baremetaldefaults.ProvisioningBridge,
+			},
+			Validate: survey.ComposeValidators(survey.Required, interfaceValidator),
+		},
+	}, &provisioningBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
```

Comment:
why is a default no appropriate for this.

Reply:
I don't understand what you mean by this. We have a default, although the behavior is a bit complex. In most cases, it is fine to look up these values from DNS. They're mandatory records. However, there are network architectures where the address in DNS may not be the one we want to use internally, so it is necessary to allow users to override the DNS value.

This is further complicated by the fact that when we ask platform-specific questions we do not have the cluster domain yet because on other platforms the cluster domain has a dependency on platform-specific install-config data. In order for us to look these values up from DNS we need the domain, so to break that cycle we do the lookup later while setting platform defaults.

So the default here is a placeholder that means "use the DNS value", but we don't have the DNS value ahead of time so we can't actually populate that. The same logic is used for the ingress vip below.

Reply:
The goal for terminal prompts is to ask the questions for fields, without which you cannot move forward.

And therefore my question if most users will use or be happy with an a specific IP for APIVIP and IngressVIP so that this prompt is not useful for most.
eg. if the prompt was for localhost IP, 127.0.0.1 is a good default and we don't need to ask the user that question.

and people who wan the customization can still specify by using the install-config.yaml

Reply:
I'd be OK dropping TUI support completely, personally.  I really don't expect anyone to ever use it for the `baremetal` platform.  There's too many details that need to be filled in about each bare metal host for a TUI to be a user friendly path.  I think we could just revisit the TUI later.  In this case, we're not looking at something like 5 questions, it's more like 30+.